### ISSUE TITLE
Datagrid: "No Data Available" message shows when filtering on certain dataset record count and filtered count

### ISSUE THREAD

**Issue by sharmainerivera** [infor-design/enterprise#3697]
**Describe the bug**
After filtering on a datagrid with only 2 data and retrieving 1 data, "No Data Available" message shows.

**To Reproduce**
1. Go to https://master-enterprise.demo.design.infor.com/components/datagrid/example-filter.html
2. Remove other rows and retain only 2 rows.
![image](https://user-images.githubusercontent.com/61222084/78231520-84b59380-7505-11ea-9562-ebafc8ddd1f6.png)
3. Filter for a certain row to be retrieved.
![image](https://user-images.githubusercontent.com/61222084/78231596-a31b8f00-7505-11ea-9e90-9e90cd09bbea.png)

**Expected behavior**
"No Data Available" should not show since there is a retrieved data.

**Version**
- ids-enterprise: 4.27.0-beta.0

**Platform**
 - OS Version: Windows 10
 - Browser Name: Chrome
 - Browser Version: Version 80.0.3987.162 (Official Build) (64-bit)

**Comment by sharmainerivera** [comment]
To add, this also happens on lookups. To replicate, we made the dataset into 6 and added the word "test" on the first three to filter them out.

![image](https://user-images.githubusercontent.com/61222084/78638531-01b68380-78df-11ea-8343-b50dbff097bf.png)

Then filtered the word "test", the No Data Available message shows.
![image](https://user-images.githubusercontent.com/61222084/78638568-1266f980-78df-11ea-8d50-59aa3940df61.png)

I tried debugging where it is failing, it seems the checkEmptyMessage on datagrid.js values of readCount and filteredCount are equals which should not be.

![image](https://user-images.githubusercontent.com/61222084/78638654-375b6c80-78df-11ea-8823-bd0fae0e01ee.png)

Kindly advise if this needs a separate ticket. Thanks!
